src/foundations/ui/slider/slider.tsx

```diff
@@ -0,0 +1,201 @@
+    "[&:focus::-ms-thumb]:ring-gray-400/50",
+  ],
+  variants: {
+    variant: {
+      horizontal: [
+        "w-full h-[var(--track-thickness)]",
+
+        // track
+        // webkit (chrome, safari)
+        "[&::-webkit-slider-runnable-track]:h-[var(--track-thickness)]",
+        // firefox
+        "[&::-moz-range-track]:h-[var(--track-thickness)]",
+        // ie + edge
+        "[&::-ms-track]:h-[var(--track-thickness)]",
+      ],
+      vertical: [
+        "w-[var(--track-thickness)] h-full [writing-mode:vertical-lr] [direction:rtl]",
+
+        // track
+        // webkit (chrome, safari)
+        "[&::-webkit-slider-runnable-track]:w-[var(--track-thickness)]",
+        // firefox
+        "[&::-moz-range-track]:w-[var(--track-thickness)]",
+        // ie + edge
+        "[&::-ms-track]:w-[var(--track-thickness)]",
+      ],
+    },
+  },
+  defaultVariants: {
+    variant: "horizontal",
+  },
+});
+
+const backgroundTrackStyle = cva({
+  base: ["absolute bg-gray-800 rounded-full"],
+  variants: {
+    variant: {
+      horizontal:
+        "left-[calc(var(--thumb-size)/2)] w-[calc(100%-var(--thumb-size))] h-[var(--track-thickness)]",
+      vertical:
+        "bottom-[calc(var(--thumb-size)/2)] w-[var(--track-thickness)] h-[calc(100%-var(--thumb-size))]",
+    },
+  },
+  defaultVariants: {
+    variant: "horizontal",
+  },
+});
+
+const progressTrackStyle = cva({
+  base: ["absolute bg-gray-200 rounded-full z-[500]", "disabled:bg-gray-400"],
+  variants: {
+    variant: {
+      horizontal: "left-[calc(var(--thumb-size)/2)]",
+      vertical: "bottom-[calc(var(--thumb-size)/2)]",
+    },
+  },
+  defaultVariants: {
+    variant: "horizontal",
+  },
+});
+
+const Slider = ({
+  min = 0,
+  max = 100,
+  step = 1,
+  value,
+  onChange,
+  orientation = "horizontal",
+  disabled = false,
+  ...props
+}: SliderProps) => {
+  const calcProgressTrackFactor = (value: number) => value / max;
+
+  return (
+    <div
+      className={cn(
+        containerStyle({ variant: orientation }),
+        disabled && "pointer-events-none opacity-50"
+      )}
+      style={
+        {
+          "--progress-track-factor": `${calcProgressTrackFactor(value)}`,
+        } as React.CSSProperties
+      }
+    >
+      <div className={backgroundTrackStyle({ variant: orientation })} />
+      <div
+        className={progressTrackStyle({ variant: orientation })}
+        style={{
+          width:
+            orientation === "horizontal"
+              ? `calc(var(--progress-track-factor)*100% - var(--thumb-size) * var(--progress-track-factor))`
+              : "var(--track-thickness)",
+          height:
+            orientation === "vertical"
+              ? `calc(var(--progress-track-factor)*100% - var(--thumb-size) * var(--progress-track-factor))`
+              : "var(--track-thickness)",
+        }}
+      />
+      <input
+        type="range"
+        min={min}
+        max={max}
+        step={step}
+        defaultValue={value}
```

Comment:
```suggestion
      <input
        type="range"
        min={min}
        max={max}
        step={step}
        defaultValue={value}
        disabled={disabled}
```

Add the `disabled` prop to the input